In [1]:
import pandas as pd

In [2]:
df = pd.read_excel("../data/datamodel_pizzasales.xlsx")

In [3]:
df.head()

order_details_id  order_id       pizza_id  quantity order_date order_time  \
0                 1         1     hawaiian_m         1 2015-01-01   11:38:36   
1                 2         2  classic_dlx_m         1 2015-01-01   11:57:40   
2                 3         2  five_cheese_l         1 2015-01-01   11:57:40   
3                 4         2    ital_supr_l         1 2015-01-01   11:57:40   
4                 5         2     mexicana_m         1 2015-01-01   11:57:40   

   unit_price  total_price pizza_size pizza_category  \
0       13.25        13.25          M        Classic   
1       16.00        16.00          M        Classic   
2       18.50        18.50          L         Veggie   
3       20.75        20.75          L        Supreme   
4       16.00        16.00          M         Veggie   

                                   pizza_ingredients  \
0           Sliced Ham, Pineapple, Mozzarella Cheese   
1  Pepperoni, Mushrooms, Red Onions, Red Peppers,...   
2  Mozzarella Cheese, Provolone Cheese, Smoked Go...   
3  Calabrese Salami, Capocollo, Tomatoes, Red Oni...   
4  Tomatoes, Red Peppers, Jalapeno Peppers, Red O...   

                  pizza_name  
0         The Hawaiian Pizza  
1   The Classic Deluxe Pizza  
2      The Five Cheese Pizza  
3  The Italian Supreme Pizza  
4         The Mexicana Pizza

# General Statistics and Data Types

In [4]:
df.shape

(48620, 12)

In [10]:
df.describe().T

count          mean           std   min       25%  \
order_details_id  48620.0  24310.500000  14035.529381  1.00  12155.75   
order_id          48620.0  10701.479761   6180.119770  1.00   5337.00   
quantity          48620.0      1.019622      0.143077  1.00      1.00   
unit_price        48620.0     16.494132      3.621789  9.75     12.75   
total_price       48620.0     16.821474      4.437398  9.75     12.75   
hour              48620.0     15.911806      3.149836  9.00     13.00   

                      50%       75%       max  
order_details_id  24310.5  36465.25  48620.00  
order_id          10682.5  16100.00  21350.00  
quantity              1.0      1.00      4.00  
unit_price           16.5     20.25     35.95  
total_price          16.5     20.50     83.00  
hour                 16.0     18.00     23.00

In [11]:
df.describe(include = "object").T

count unique  \
pizza_id           48620     91   
pizza_size         48620      5   
pizza_category     48620      4   
pizza_ingredients  48620     32   
pizza_name         48620     32   

                                                                 top   freq  
pizza_id                                                  big_meat_s   1811  
pizza_size                                                         L  18526  
pizza_category                                               Classic  14579  
pizza_ingredients  Pepperoni, Mushrooms, Red Onions, Red Peppers,...   2416  
pizza_name                                  The Classic Deluxe Pizza   2416

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48620 entries, 0 to 48619
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   order_details_id   48620 non-null  int64         
 1   order_id           48620 non-null  int64         
 2   pizza_id           48620 non-null  object        
 3   quantity           48620 non-null  int64         
 4   order_date         48620 non-null  datetime64[ns]
 5   order_time         48620 non-null  object        
 6   unit_price         48620 non-null  float64       
 7   total_price        48620 non-null  float64       
 8   pizza_size         48620 non-null  object        
 9   pizza_category     48620 non-null  object        
 10  pizza_ingredients  48620 non-null  object        
 11  pizza_name         48620 non-null  object        
dtypes: datetime64[ns](1), float64(2), int64(3), object(6)
memory usage: 4.5+ MB


In [7]:
df.isnull().sum()

order_details_id     0
order_id             0
pizza_id             0
quantity             0
order_date           0
order_time           0
unit_price           0
total_price          0
pizza_size           0
pizza_category       0
pizza_ingredients    0
pizza_name           0
dtype: int64

The dataframe consists of 48620 records and 12 variables. Most of the variables are numerical, and there are no missing values. The average price of a pizza is $16.5, with 32 unique pizza types. A more in-depth analysis will be performed using Tableau.

# Formating and Cleaning Data

In [12]:
# Converting the "order_time" column from its current data type to a datetime data type
df["order_time"] = pd.to_datetime(df["order_time"], format="%H:%M:%S")

# Creating a new "hour" column to store the hour component of the "order_time"
df["hour"] = df["order_time"].dt.hour

# Counting the number of orders for each hour and sorting the results in ascending order of the hour
df["hour"].value_counts().sort_index()

9        4
10      17
11    2672
12    6543
13    6203
14    3521
15    3170
16    4185
17    5143
18    5359
19    4350
20    3487
21    2528
22    1370
23      68
Name: hour, dtype: int64

In [9]:
# Creating a new dataframe from the "pizza_name" and "pizza_ingredients" columns
# to obtain unique values for "pizza_name"
ingredients = df[["pizza_name", "pizza_ingredients"]].copy()
ingredients = ingredients.drop_duplicates(subset=["pizza_name"])

# Splitting the "pizza_ingredients" column by comma and exploding the resulting list
# to identify individual ingredients used in each type of pizza
ingredients["pizza_ingredients"] = ingredients["pizza_ingredients"].str.split(',')
ingredients = ingredients.explode("pizza_ingredients")

# Checking the resulting dataframe
ingredients.head()


pizza_name   pizza_ingredients
0        The Hawaiian Pizza          Sliced Ham
0        The Hawaiian Pizza           Pineapple
0        The Hawaiian Pizza   Mozzarella Cheese
1  The Classic Deluxe Pizza           Pepperoni
1  The Classic Deluxe Pizza           Mushrooms

# Exporting the dataframes to .csv format for storage

In [13]:
df.to_csv('../data/pizza_sales.csv', index=False)

In [14]:
ingredients.to_csv('../data/ingredients.csv', index=False)